include/NovelRT/Graphics/TextRect.h

```diff
@@ -1,50 +1,51 @@
 // Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
 
-#ifndef NOVELRT_NOVELTEXTRECT_H
-#define NOVELRT_NOVELTEXTRECT_H
-#include "NovelRenderObject.h"
-#include "RGBAConfig.h"
-#include <string>
-#include "GraphicsCharacterRenderData.h"
-#include "NovelImageRect.h"
-#include "NovelLoggingService.h"
-#include "ShaderProgram.h"
-
-namespace NovelRT {
-  class NovelTextRect : public NovelRenderObject {
+#ifndef NOVELRT_GRAPHICS_TEXTRECT_H
+#define NOVELRT_GRAPHICS_TEXTRECT_H
+#ifndef NOVELRT_H
+#error Please do not include this directly. Use the centralised header (NovelRT.h) instead!
+#endif
+
+namespace NovelRT::Graphics {
+  class TextRect : public RenderObject {
   private:
     void reloadText();
 
     std::string _fontFileDir;
     std::string _previousFontFileDir = "";
     std::string _text = "";
-    std::vector<NovelImageRect*> _letterRects;
+    std::vector<ImageRect*> _letterRects;
     std::map<GLchar, GraphicsCharacterRenderData> _fontCharacters;
-    NovelCommonArgs _args;
-    NovelLoggingService _logger;
+    LoggingService _logger;
     RGBAConfig _colourConfig;
     float _fontSize;
 
   protected:
     void configureObjectBuffers() final;
 
   public:
-    NovelTextRect(NovelLayeringService* layeringService,
-      const NovelCommonArgs& args,
+    TextRect(const Transform& transform,
+      int layer,
       ShaderProgram programId,
-      NovelCamera* camera,
+      Camera* camera,
       float fontSize,
       const std::string& fontFileDir,
       const RGBAConfig& colourConfig);
 
     void drawObject() final;
 
-    RGBAConfig getColourConfig() const;
+    inline const RGBAConfig& getColourConfigReadonly() const {
```

Comment:
I don't think you need to call this `Readonly`

Reply:
Oh, alright. Then what do we call it? I thought it needed a different name or the compiler would screech. I also like the clarity.

FWIW, I have used this in other places for consistency.

Reply:
You can just give it the same name and the compiler picks the correct overload for the use-case.
Types in the stdlib do the same thing: <http://www.cplusplus.com/reference/vector/vector/at/>

Reply:
Alright, then we'll do this if you believe this to be clearer. If the stdlib does it then I assume people know what they can and can't do based on the context of being `const` or not.